src/aws-cpp-sdk-core/include/smithy/identity/signer/built-in/SigV4aSigner.h

```diff
@@ -0,0 +1,202 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+#pragma once
+
+#include <smithy/identity/signer/AwsSignerBase.h>
+#include <smithy/identity/identity/AwsCredentialIdentityBase.h>
+
+//#include <aws/core/auth/signer/AWSAuthV4Signer.h>
+
+#include <aws/core/auth/AWSCredentials.h>
+#include <aws/crt/auth/Credentials.h>
+
+#include <aws/core/http/HttpRequest.h>
+#include <aws/core/auth/signer/AWSAuthSignerHelper.h>
+#include <aws/crt/http/HttpConnection.h>
+#include <aws/crt/http/HttpRequestResponse.h>
+
+
+namespace smithy {
+    static const char* UNSIGNED_PAYLOAD = "UNSIGNED-PAYLOAD";
+    static const char* EMPTY_STRING_SHA256 = "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855";
+    static const char v4AsymmetricLogTag[] = "AWSAuthSymmetricV4Signer";
+    static const char* USER_AGENT = "user-agent";
+
+
+    /**
+     * A smithy SigV4 signer wrapper on top of legacy SDK SigV4 signer
+     * TODO: refactor into own signer using smithy design
+     */
+    class AwsSigV4aSigner : public AwsSignerBase<AwsCredentialIdentityBase> {
+        
+    public:
+        using SigV4aAuthSchemeParameters = smithy::DefaultAuthSchemeResolverParameters;
+        explicit AwsSigV4aSigner(const SigV4aAuthSchemeParameters& parameters)
+            :  m_parameters{parameters}
+        {
+        }
+
+        SigningFutureOutcome sign(std::shared_ptr<HttpRequest> httpRequest, const AwsCredentialIdentityBase& identity, SigningProperties properties) override
```

Comment:
please note that this uses legacy implementation just refactored and cleaned up a bit